<?xml version="1.0"?>
<launch>

    <param name="use_sim_time" value="true"/>
    <!-- launch your velodyne here -->
    <!-- <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/> -->
    
    <!-- For Velodyne VLP-16 -->
    <!-- <param name="scan_line" value="16" /> -->
    

    <!-- For Velodyne HDL-32 
    <param name="scan_line" value="32" />
    -->

    <!-- For Velodyne HDL-64  -->
    <param name="scan_line" value="64" />
   

    <!--- Sim Time -->
    <param name="scan_period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="map_resolution" type="double" value="0.2" />
    <param name="max_dis" type="double" value="90.0" />
    <param name="min_dis" type="double" value="0.5" />

    <arg name="output_folder" default="$(find lcloam)/test/dt/"/>
    <param name="dt_folder" type="string" value="$(arg output_folder)"/>

    <!--- -->
    <node pkg="lcloam" type="lcloam_odom_estimation_node" name="lcloam_odom_estimation_node" output="screen"/>
    <node pkg="lcloam" type="lcloam_laser_processing_node" name="lcloam_laser_processing_node" output="screen"/>
    <node pkg="lcloam" type="lcloam_backend_opt_node" name="lcloam_backend_opt_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />

    <rosparam command="load" file="$(find lcloam)/config/params.yaml" />

</launch>
